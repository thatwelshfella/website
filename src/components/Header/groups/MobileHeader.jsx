import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { FaBars, FaChevronRight } from 'react-icons/fa';

import '../Header.less';

const MobileHeader = ({ icon, menu }) => {
  const [showMenu, setShowMenu] = useState(false);

  const toggleMobileMenu = () => {
    setShowMenu(!showMenu);
  };

  return (
    <>
      <div className="header-component">
        {icon}
        <FaBars onClick={toggleMobileMenu} />
      </div>
      <div className={showMenu ? 'mobile-menu show' : 'mobile-menu hidden'}>
        <div
          className={
            showMenu ? 'menu-items container' : 'menu-items container hidden'
          }>
          {menu.map(items => (
            <a href={items.link} onClick={toggleMobileMenu} key={items.link}>
              {items.value.toUpperCase()}
              {`  `}
              <FaChevronRight />
            </a>
          ))}
        </div>
      </div>
    </>
  );
};

MobileHeader.propTypes = {
  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  menu: PropTypes.array.isRequired,
};

export default MobileHeader;
