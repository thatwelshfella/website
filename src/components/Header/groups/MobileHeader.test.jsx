import React from 'react';
import { shallow } from 'enzyme';
import MobileHeader from './MobileHeader';

describe('<MobileHeader />', () => {
  const menu = [
    { link: '#home', value: 'Home' },
    { link: '#aboutMe', value: 'About me' },
    { link: '#timeline', value: 'Employment Timeline' },
    { link: '#github', value: 'Github Projects' },
  ];

  const cmp = shallow(<MobileHeader icon="LD" menu={menu} />);

  describe('@renders', () => {
    it('should render with mobile header ', () => {
      expect(cmp).toMatchSnapshot();
    });
  });

  describe('@actions', () => {
    it('should toggle open the mobile menu', () => {
      const menu = cmp.find('FaBars').first();

      menu.simulate('click');

      expect(cmp.find('[className="mobile-menu show"]').length).toEqual(1);
    });
  });
});
