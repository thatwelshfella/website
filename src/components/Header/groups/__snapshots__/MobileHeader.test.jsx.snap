// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileHeader /> @renders should render with mobile header  1`] = `
<Fragment>
  <div
    className="header-component"
  >
    LD
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="bars"
      inverse={false}
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size="lg"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <div
    className="mobile-menu hidden"
  >
    <div
      className="menu-items container hidden"
    >
      <a
        href="#home"
        key="#home"
        onClick={[Function]}
      >
        HOME
          
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="chevron-right"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
      <a
        href="#aboutMe"
        key="#aboutMe"
        onClick={[Function]}
      >
        ABOUT ME
          
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="chevron-right"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
      <a
        href="#timeline"
        key="#timeline"
        onClick={[Function]}
      >
        EMPLOYMENT TIMELINE
          
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="chevron-right"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
      <a
        href="#github"
        key="#github"
        onClick={[Function]}
      >
        GITHUB PROJECTS
          
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="chevron-right"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
    </div>
  </div>
</Fragment>
`;
